soda_machine.rb

```diff
@@ -7,12 +9,26 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      else
+        return nil
+      end
+    end
   end
 
   def sell(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        @cash += soda.price
+        @sodas.delete(soda)
+      end
+    end
+    return nil
```

Comment:
This is needed, because otherwise you return whatever is on the last line, which would be the result of `@sodas.each` which will be the sodas array itself!

Note that the return value is only checked in the tests in cases where the item isn't there -- ideally you'd return the soda in the other case.